cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -177,6 +157,15 @@ type ImageMeta struct {
 	//TODO: evaluate if we need also a ImageName based on user feedbacks
 }
 
+// AddOn defines settings to be used when deploying a specific addon kind
+type AddOn struct {
+	// Kind defines the addon type
+	Kind string
```

Comment:
> Kind defines the addon type

should this be
```
Type AddonType
```

a courtesy of:
```
 	// Type defines the DNS add-on to be used
	Type DNSAddOnType
```

on Addon vs AddOn i'm in favor of using the former.
the correct spelling in english is "add-on", but kubernetes has usage of "addon" in a lot of places and it's not invalid spelling.

Reply:
Might be I'm missing something, but If this should be an enum, then how the user is expected to pass any addon different from what is the list?

Generalizing, How this is expected to translate into

```
apiVersion: addons.config.k8s.io/v1alpha1
kind: AddonInstallerConfiguration
addons:
- name: kube-proxy
  ref: github.com/kubernetes/kubernetes//cluster/addons/kustomize/kube-proxy/?ref=v1.17.0
```

Reply:
> Might be I'm missing something, but If this should be an enum, then how the user is expected to pass any addon different from what is the list?

my impressions is that this feature was targeting only core kubeadm addons.
for non-core addons users can always `kubeactl apply` as a post step.

Reply:
At this point this is a rough translation of what we have into what we need.
My idea here is to allow for a few kubeadm concerned settings to be tweaked (currently image tag and repo). The addon manager and addon deployment code will do the rest.

Reply:
P.S. This is deliberately not going to be an enum. We don't know what addons are going to be supported by the addons installer, hence we don't want to restrict the user here.

Reply:
commented on another location in the PR, IMO/AFAIK we would want the addon installer to not embed it's config in the kubeadm config.